public/main/inc/lib/webservices/Rest.php

```diff
@@ -1432,6 +1435,97 @@ public function subscribeUserToCourse($params)
+                    $row["url"]=$url;
+                    $homeWork[] = $row; 
+                }
+                return $homeWork;
+        }          
+        return [];
+    }
+    /**
+     * @param $course_id
+     * @param $homeWork_id
+     * @return array
+     */
+    public function getInformationHomework($course_id, $homeWork_id){
+        $workTable =Database::get_course_table(TABLE_STUDENT_PUBLICATION);
+        $workassignment =Database::get_course_table(TABLE_STUDENT_PUBLICATION_ASSIGNMENT);
+        $workcomment =Database::get_course_table(TABLE_STUDENT_PUBLICATION_ASSIGNMENT_COMMENT);
+        $idUser=$this->user->getId();
+        $informationWork= "SELECT *,w.id AS id_w FROM $workTable as w LEFT JOIN $workassignment AS a 
+                            ON a.publication_id=w.id WHERE w.c_id = $course_id AND w.parent_id = $homeWork_id 
+                            AND w.user_id=$idUser;";
+        $informationHomework=[];
+        $result = Database::query($informationWork);
+        $course_code=self::getCourseCode($course_id); 
+        if (Database::num_rows($result)) {
+            while($row= Database::fetch_array($result, 'ASSOC')){
+                $id=$row["id_w"];
+                $studentId = $row['user_id'];
+                $teacherId = $row['qualificator_id'];
+                $url="https://".$_SERVER['SERVER_NAME']."/main/work/view.php?cidReq=$course_code&id_session=0&gidReq=0&gradebook=0&origin=&id=$id";
+                $row['url'] = $url;
+                
+                //Feedback is obtained on each task submitted by the specific user
+                 $commentWork= "SELECT c_c.*, concat(u.lastname,' ',u.firstname) as user_comment
+                                FROM $workcomment AS c_c 
+                                INNER JOIN user AS u ON c_c.user_id = u.user_id
+                                WHERE work_id=$id AND (c_c.user_id=$studentId OR c_c.user_id=$teacherId)";
+                 $resultComment = Database::query($commentWork);
+                 if (Database::num_rows($resultComment)) {
+                     while($rowComments= Database::fetch_array($resultComment, 'ASSOC')){
+                         //The html tags are removed from the comment and at the same time the comment is inserted with a <br> every 85 characters
+                         $rowComments['comment'] = strip_tags($rowComments['comment']);
+                         $rowComments['comment'] = wordwrap($rowComments['comment'] , 85, "<br/>",1);
+                         $row['comments'][] = $rowComments;
+                     }
+                 }else{
+                    $row['comments']['no_comment'] = "No comment";
+                 }
+                 $informationHomework[] = $row;
+            }
+        }
+        return $informationHomework;
+    }
+    
+    /**
+     * Get Course Code
+     * @param $course_id
+     */
+    private function getCourseCode($course_id)
```

Comment:
There's no need to re-implement this methos (api_get_course_info_by_id($course_id) would give you an array with a 'code')